tests/persistSync.test.tsx

```diff
@@ -0,0 +1,173 @@
+import create from '../src/index'
+import { persist } from '../src/middleware'
+
+const consoleError = console.error
+afterEach(() => {
+  console.error = consoleError
+})
+
+it('can rehydrate state', () => {
+  let postRehydrationCallbackCallCount = 0
+
+  const storage = {
+    getItem: (name: string) =>
+      JSON.stringify({
+        state: { count: 42, name },
+        version: 0,
+      }),
+    setItem: () => {},
+  }
+
+  const useStore = create(
+    persist(
+      () => ({
+        count: 0,
+        name: 'empty',
+      }),
+      {
+        name: 'test-storage',
+        getStorage: () => storage,
+        onRehydrateStorage: () => (state, error) => {
+          postRehydrationCallbackCallCount++
+          expect(error).toBeUndefined()
+          expect(state?.count).toBe(42)
+          expect(state?.name).toBe('test-storage')
+        },
+      }
+    )
+  )
+
+  expect(useStore.getState()).toEqual({ count: 42, name: 'test-storage' })
+  expect(postRehydrationCallbackCallCount).toBe(1)
+})
+
+it('can throw rehydrate error', () => {
+  let postRehydrationCallbackCallCount = 0
+
+  const storage = {
+    getItem: () => {
+      throw new Error('getItem error')
+    },
+    setItem: () => {},
+  }
+
+  create(
+    persist(() => ({ count: 0 }), {
+      name: 'test-storage',
+      getStorage: () => storage,
+      onRehydrateStorage: () => (_, e) => {
+        postRehydrationCallbackCallCount++
+        expect(e?.message).toBe('getItem error')
+      },
+    })
+  )
+
+  expect(postRehydrationCallbackCallCount).toBe(1)
+})
+
+it('can persist state', () => {
+  let setItemCallCount = 0
+
+  const storage = {
+    getItem: () => null,
+    setItem: (name: string, value: string) => {
+      setItemCallCount++
+      expect(name).toBe('test-storage')
+      expect(value).toBe(
+        JSON.stringify({
+          state: { count: 42 },
+          version: 0,
+        })
+      )
+    },
+  }
+
+  const useStore = create<any>(
```

Comment:
Maybe using real interfaces instead of `any` could help to test the typing integration? (I'm speaking of every test in this file, not this `create` specifically)

Reply:
Just removing the `<any>` will make TS automatically figure out the type correctly. Would that be sufficient?

Reply:
sure :+1: